### DISCUSSION TITLE
Adjust hue, saturation and brightness in different regions.

### DISCUSSION THREAD

**Discussion by charles1e62** [ImageMagick/ImageMagick#3654] [Help]
Is it possible to adjust hue, saturation and brightness in different regions of an image?
For example on a 200x200 image I would like to use -modulate on the regions 0,0 200,66 and 0,133.2 200,200 with different h.s.l.

**Comment by snibgo** [comment]
See https://imagemagick.org/script/command-line-options.php#region

For example:
```
magick in.png -region 100x100+50+50 -modulate 50,0,100 +region out.png
```

**Comment by charles1e62** [comment]
This command works.

``
convert 9.png \
  -set option:modulate:colorspace hsl \
  -region 200x66.6+0+0 -modulate 70,50,80 \
  -region 200x66.6+0+133.2 -modulate 70,50,80 \
  o.png
``

Can you tell me why it is producing different colors when the h.s.l parameters are the same?

![9](https://user-images.githubusercontent.com/65585831/117687507-3c4bb900-b18e-11eb-8b8a-0baa1f5ab880.png)
![o](https://user-images.githubusercontent.com/65585831/117687510-3d7ce600-b18e-11eb-98f4-7a4e822a28b3.png)

**Reply by fmw42** [reply]
-region in IM 7 does not behave as it used to in IM 6. I would suggest not using it. See https://imagemagick.org/script/porting.php#cli. Use masks in stead.

**Reply by snibgo** [reply]
@fmw42: One region works fine, but two regions don't work even when separated by "+region". Is that the expected behaviour? Or a bug?

**Reply by fmw42** [reply]
I do not know what the behavior should be in IM 7, since it has been changed.  I do not think it should be used in IM 7. You can draw rectangular masks at used them to mix the original and the processed image and do that multiply times in the command line to achieve what you want.

**Reply by charles1e62** [reply]
Thanks for the help.
This command works the way I expected.

``
convert 9.png \
  \( -clone 0 -region '200x66.6+0+0' -modulate 70,50,80 \) \
  \( -clone 1 -region '200x66.6+0+133.2' -modulate 90,50,80 \) \
  -layers merge -strip o.png
``

I tried not using -region but the other options like -geometry and -repage didn't work.

**Reply by fmw42** [reply]
Width, Heights and Offsets will be truncated or rounded to integers. Your fractions will not be used.
